### ISSUE TITLE
Main branch build failing with gcc 11.4.0

### ISSUE THREAD

**Issue by sanchitintel** [pytorch/vision#8339]
### 🐛 Describe the bug

Build failing with gcc 11.4.0 on Ubuntu 22.04 -
Commit used - `ef2920cc80bac61282b3b19a775b3c33de4e7551`

```
gcc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPNG_FOUND=1 -DJPEG_FOUND=0 -DNVJPEG_FOUND=0 -Ivision/torchvision/csrc -I/usr/include/libpng16 -Ivision/torchvision/csrc -Ipytorch/torch/include -Ipytorch/torch/include/torch/csrc/api/include -Ipytorch/torch/include/TH -Ipytorch/torch/include/THC -Ivision/torchvision/csrc/io/image -Ipytorch/torch/include -Ipytorch/torch/include/torch/csrc/api/include -Ipytorch/torch/include/TH -Ipytorch/torch/include/THC -Ienvs/pytorch/include -I/usr/local/include/python3.11 -c vision/torchvision/csrc/io/image/cpu/decode_jpeg.cpp -o build/temp.linux-x86_64-cpython-311vision/torchvision/csrc/io/image/cpu/decode_jpeg.o -g0 -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1016\" -DTORCH_EXTENSION_NAME=image -D_GLIBCXX_USE_CXX11_ABI=1 -std=c++17
In file included from vision/torchvision/csrc/io/image/cpu/decode_jpeg.cpp:3:
vision/torchvision/csrc/io/image/cpu/exif.h:200:40: error: ‘j_decompress_ptr’ was not declared in this scope
  200 | inline int fetch_jpeg_exif_orientation(j_decompress_ptr cinfo) {
      |                                        ^~~~~~~~~~~~~~~~
error: command '/usr/bin/gcc' failed with exit code 1
```

### Versions

```
PyTorch version: 2.4.0a0+git6b33e58
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.1 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: 17.0.6 (++20231209124227+6009708b4367-1~exp1~20231209124336.77)
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.11.6 (main, Jan 25 2024, 10:16:18) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-5.15.0-53-generic-x86_64-with-glibc2.35
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Address sizes:                   46 bits physical, 48 bits virtual
Byte Order:                      Little Endian
CPU(s):                          208
On-line CPU(s) list:             0-207
Vendor ID:                       GenuineIntel
Model name:                      Intel(R) Xeon(R) Platinum 8371HC CPU @ 3.30GHz
CPU family:                      6
Model:                           85
Thread(s) per core:              2
Core(s) per socket:              26
Socket(s):                       4
Stepping:                        11
CPU max MHz:                     4200.0000
CPU min MHz:                     1200.0000
BogoMIPS:                        6600.00
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local avx512_bf16 dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req pku ospke avx512_vnni md_clear flush_l1d arch_capabilities
Virtualization:                  VT-x
L1d cache:                       3.3 MiB (104 instances)
L1i cache:                       3.3 MiB (104 instances)
L2 cache:                        104 MiB (104 instances)
L3 cache:                        143 MiB (4 instances)
NUMA node(s):                    4
NUMA node0 CPU(s):               0-25,104-129
NUMA node1 CPU(s):               26-51,130-155
NUMA node2 CPU(s):               52-77,156-181
NUMA node3 CPU(s):               78-103,182-207
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Retbleed:          Mitigation; Enhanced IBRS
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] torch==2.4.0a0+git6b33e58
[conda] Could not collect
```

**Comment by sanchitintel** [comment]
Had resolved it locally by installing `libjpeg`:

```
apt install libjpeg-dev
```

**Comment by sanchitintel** [comment]
Will be fixed by #8342
